lib/style/defs.ex

```diff
@@ -41,60 +41,107 @@ defmodule Styler.Style.Defs do
 
   @behaviour Styler.Style
 
+  alias Styler.Style
   alias Styler.Zipper
 
   # a def with no body like
   #
   #  def example(foo, bar \\ nil)
   #
   def run({{def, meta, [head]}, _} = zipper, ctx) when def in [:def, :defp] do
+    {_fn_name, head_meta, _children} = head
+    first_line = meta[:line]
+    last_line = head_meta[:closing][:line]
+
+    comments =
+      if first_line == last_line do
+        ctx.comments
+      else
+        Style.displace_comments(ctx.comments, first_line..last_line)
+      end
```

Comment:
i think you can use this branch to just skip all work, not just moving coments!
first clause is `{:skip, zipper, ctx)`, second line is all the displace/flatten/replace work